File: scryfall/ruling.go
package scryfall

import "time"

type Ruling struct {
	Object      string `json:"object"`
	OracleID    string `json:"oracle_id"`
	Source      string `json:"source"`
	PublishedAt string `json:"published_at"`
	Comment     string `json:"comment"`
}

func (r *Ruling) PublishedAtTime() time.Time {
	return TimestampToTime(r.PublishedAt)
}

File: packages/scryfall/src/types/ruling.ts
import {ScryfallDate} from './scryfall';

export type Ruling = {
    object: 'ruling';
    oracle_id: string;
    source: 'scryfall' | 'wotc';
    published_at: ScryfallDate;
    comment: string;
};

export function isRuling(o: unknown): o is Ruling {
    return Boolean(o && typeof o === 'object' && (o as any).object === 'ruling');
}
